evm-contracts/src/v0.6/dev/DeviationFlaggingValidator.sol

```diff
@@ -0,0 +1,166 @@
+pragma solidity ^0.6.0;
+
+import './AggregatorValidatorInterface.sol';
+import '../interfaces/FlagsInterface.sol';
+import '../Owned.sol';
+import '../vendor/SafeMath.sol';
+import '../SignedSafeMath.sol';
+
+/**
+ * @title The Deviation Flagging Validator contract
+ * @notice Checks the current value against the previous value, and makes sure
+ * that it does not deviate outside of some relative range. If the deviation
+ * threshold is passed then the validator raises a flag on the designated
+ * flag contract.
+ */
+contract DeviationFlaggingValidator is Owned, AggregatorValidatorInterface {
+  using SafeMath for uint256;
+  using SignedSafeMath for int256;
+
+  uint32 constant public THRESHOLD_MULTIPLIER = 100000;
+
+  uint32 public flaggingThreshold;
+  FlagsInterface public flags;
+
+  event FlaggingThresholdUpdated(
+    uint24 indexed previous,
+    uint24 indexed current
+  );
+  event FlagsAddressUpdated(
+    address indexed previous,
+    address indexed current
+  );
+
+  /**
+   * @notice sets up the validator with its threshold and flag address.
+   * @param _flags sets the address of the flags contract
+   * @param _flaggingThreshold sets the threshold that will trigger a flag to be
+   * raised. Setting the value of 100,000 is equivalent to tolerating a 100%
+   * change compared to the previous price.
+   */
+  constructor(
+    address _flags,
+    uint24 _flaggingThreshold
+  )
+    public
+  {
+    setFlagsAddress(_flags);
+    setFlaggingThreshold(_flaggingThreshold);
+  }
+
+  /**
+   * @notice checks whether the parameters count as valid by comparing the
+   * difference change to the flagging threshold.
+   * @param previousRoundId is ignored.
+   * @param previousAnswer is used as the median of the difference with the
+   * current answer to determine if the deviation threshold has been exceeded.
+   * @param roundId is ignored.
+   * @param answer is the latest answer which is compared for a ratio of change
+   * to make sure it has not execeeded the flagging threshold.
+   */
+  function validate(
+    uint256 previousRoundId,
```

Comment:
We again mix naming conventions in the same file. Some function params here have an underscore in front, others do not. We should use one convention for all functions.

Reply:
This is exactly why I prefer the leading underscore. We already use it in almost all of our contracts.

Reply:
I've updated them all to use underscores. Long term, I think we should more towards explicitly denoting storage variables with a prefix like `stored` so we can be make gas usage standout, then we won't need to denote parameters with an underscore.

Reply:
Cool, or maybe long term we can shorten that to `s_` and `m_` since we tend to create local memory variables a lot as well.